lib/src/database/db.dart

```diff
@@ -209,7 +209,12 @@ class Db {
       }
 
       if (connection == null) {
-        connection = _masterConnection;
+        if (_masterConnection != null) {
+          connection = _masterConnection;
+        }
+        else {
+          throw new MongoDartError('No master connection');
```

Comment:
I don't know if it'd be easy to write a test for this as well, but if you think it's not too hard, I'd like to see one in there. Thanks for the contribution btw.